runtime/doc/lsp.txt

```diff
@@ -158,6 +158,27 @@ name: >
   vim.lsp.protocol.TextDocumentSyncKind.Full == 1
   vim.lsp.protocol.TextDocumentSyncKind[1] == "Full"
 
+================================================================================
+LSP HIGHLIGHT GROUPS                                      *lsp-highlight-groups*
+
+                                                        *hl-LspDiagnosticsError*
+LspDiagnosticsError       used for 'Error' diagnostic virtual text
+                                                      *hl-LspDiagnosticsWarning*
+LspDiagnosticsWarning     used for 'Warning' diagnostic virtual text
+                                                  *hl-LspDiagnosticsInformation*
+LspDiagnosticInformation  used for 'Information' diagnostic virtual text
+                                                         *hl-LspDiagnosticsHint*
+LspDiagnosticHint         used for 'Hint' diagnostic virtual text
+                                                    *hl-LspDiagnosticsUnderline*
+LspDiagnosticsUnderline   used for text with diagnostic information
+                                                           *hl-LspReferenceText*
+LspReferenceText          used for highlighting 'text' references
+                                                           *hl-LspReferenceRead*
+LspReferenceRead          used for highlighting 'read' references
+                                                          *hl-LspReferenceWrite*
+LspReferenceWrite         used for highlighting 'write' references
```

Comment:
where is all this stuff actually used? Also do we really need all of this ?

Reply:
This is used by the existing code, eg: https://github.com/neovim/neovim/blob/master/runtime/lua/vim/lsp/util.lua#L564

the only ones that can be merged in my opinion are LspReferenceRead, LspReferenceWrite and LspReferenceText, but users may want to differentiate them